### ISSUE TITLE
Cannot build flutter_tools on Linux, Because flutter_tools depends on collection >=1.15.0-nnbd <1.15.0-nullsafety.2 which requires SDK version >=2.9.0-18.0 <=2.9.10, version solving failed.

### ISSUE THREAD

**Issue by jmagman** [flutter/flutter#62146]
UPDATE: This issue has been mitigated in the framework by rolling back the engine hash to an unaffected version. However, the latest engine build still has the wrong dart version: https://ci.chromium.org/p/flutter/builders/prod/Linux%20Host%20Engine/6083. I suspect we will have to delete the git caches of all the LUCI bots.

------

Building flutter tools is failing on linux
https://cirrus-ci.com/task/4955835350646784?command=setup#L337
```
flutter config --no-analytics
Downloading Dart SDK from Flutter engine dcc9a4048da890660f0880b4c2afed7c9470d335...
```
```
The current Dart SDK version is 2.10.0-0.0.dev.flutter-9dca49e71e.
Because flutter_tools depends on collection >=1.15.0-nnbd <1.15.0-nullsafety.2 which requires SDK version >=2.9.0-18.0 <=2.9.10, version solving failed.
Error: Unable to 'pub upgrade' flutter tool. Retrying in five seconds... (9 tries left)
Resolving dependencies...
```
Seeing this on Cirrus and devicelab.  Windows and macOS is passing.

**Comment by jmagman** [comment]
Dart 2.10 hasn't even been rolled into the engine yet.

**Comment by zanderso** [comment]
This repros for locally at head:
```
$ ./bin/cache/dart-sdk/bin/dart --version
Dart SDK version: 2.10.0-0.0.dev.flutter-9dca49e71e (be) (Fri Jul 17 13:26:08 2020 +0000) on "linux_x64"
```

**Comment by jmagman** [comment]
On my mac where it's working, that output is:
```
$ bin/cache/dart-sdk/bin/dart --version
Dart SDK version: 2.9.0-21.0.dev.flutter-9dca49e71e (be) (Fri Jul 17 13:26:08 2020 +0000) on "macos_x64"
```
Same Dart SHA [9dca49e71e](https://github.com/dart-lang/sdk/commit/9dca49e71e), different version number.

**Comment by JheiKrauzer0613** [comment]
I got this error after I run flutter upgrade in Android Studio - Windows 10
```
Because flutter_tools depends on collection >=1.15.0-nnbd <1.15.0-nullsafety.2 which requires SDK version >=2.9.0-18.0 <=2.9.10, version solving failed.
```

Now, I can't run even a simple flutter clean. I hope this can be fix right away.

**Comment by kf6gpe** [comment]
Being proactive, assigning to @a-siva and @whesse. Can one of you please have a look?

**Comment by a-siva** [comment]
How did flutter's cache get Dart 2.10, The roll of Dart 2.10 into the engine has not happened yet.

**Comment by zanderso** [comment]
Is the engine build of the Dart SDK taking the version from tags in the Dart repo? Were any new tags added recently? @tvolkert

**Comment by zanderso** [comment]
Yep:
```
$ git for-each-ref refs/tags/*dev --sort=-taggerdate "--format=%(refname:strip=2)" --count=1
2.10.0-0.0.dev
```

**Comment by a-siva** [comment]
Do we still do this version tag hack, I thought it was fixed when Patrick switched to the branching strategy for Flutter

**Comment by zanderso** [comment]
I don't know, but it looks like it is probably coming from here: https://github.com/dart-lang/sdk/blob/master/tools/make_version.py#L87

**Comment by przemekbroda** [comment]
I have this same problem. When I run flutter upgrade console throws "Because flutter_tools depends on collection >=1.15.0-nnbd <1.15.0-nullsafety.2 which requires SDK version >=2.9.0-18.0 <=2.9.10, version solving failed" and it goes into infinite loop with it

**Comment by christopherfujino** [comment]
cc @athomas

**Comment by zanderso** [comment]
@a-siva @whesse can the tag be removed?

**Comment by christopherfujino** [comment]
This PR https://github.com/flutter/flutter/pull/62150/files builds for me locally

**Comment by leafpetersen** [comment]
cc @jakemac53

**Comment by a-siva** [comment]
> @a-siva @whesse can the tag be removed?

Apparently @whesse thinks it may be hard to remove the tag.

Is there an option for Flutter to not use the tag when building the version number and instead use the actual version number

**Comment by jmagman** [comment]
Why is it only happening on Linux?

**Comment by zanderso** [comment]
@a-siva, I believe the version string comes from the `make_version.py` script in the Dart repo.

**Comment by zanderso** [comment]
@christopherfujino To clarify, did the manual engine roll not have the Dart SDK version problem? @jason-simmons if it did not have the problem, do you know why it didn't have the problem?

**Comment by a-siva** [comment]
> @a-siva, I believe the version string comes from the `make_version.py` script in the Dart repo.

This was a hack added for Flutter and only flutter invokes the make_version.py script with the git-hash option. We should get rid of this hack first thing after this current problem is solved.

**Comment by christopherfujino** [comment]
> @christopherfujino To clarify, did the manual engine roll not have the Dart SDK version problem?

Correct, on PR #62150, the tool correctly builds, with:

```
[✓] Flutter (Channel unknown, 1.21.0-6.0.pre.13, on Linux, locale en_US.UTF-8)
    • Flutter version 1.21.0-6.0.pre.13 at /home/chris/git/flutter
    • Framework revision bd5d8da372 (24 minutes ago), 2020-07-23 11:06:41 -0700
    • Engine revision 4d720dd9e5
    • Dart version 2.9.0 (build 2.9.0-21.0.dev fcd1c5b61a)
```

**Comment by jakemac53** [comment]
These packages all have `.2` releases which allow 2.10, and when the sdk version number changes to `2.10` those versions of the packages need to be pinned in the pubspecs.

So for instance given the original failure text:

```
The current Dart SDK version is 2.10.0-0.0.dev.flutter-9dca49e71e.
Because flutter_tools depends on collection >=1.15.0-nnbd <1.15.0-nullsafety.2 which requires SDK version >=2.9.0-18.0 <=2.9.10, version solving failed.
Error: Unable to 'pub upgrade' flutter tool. Retrying in five seconds... (9 tries left)
Resolving dependencies...
```

The `flutter_tools` pubspec needs to change to have this constraint `>=1.15.0-nnbd <1.15.0-nullsafety.2` (this could be landed ahead of time, as it supports versions that are compatible with both 2.9 and 2.10).

**Comment by jason-simmons** [comment]
I haven't been able to reproduce exactly how LUCI is fetching the Dart repository and why the `2.10.0-0.0.dev` tag is showing up in some engine Dart SDK builds but not others.

I suspect the engine recipe might get more consistent results by doing a `git fetch --tags` instead of just a `git fetch` (see https://flutter.googlesource.com/recipes/+/refs/heads/master/recipes/engine.py#1496)

@godofredoc

**Comment by zanderso** [comment]
When I build the engine locally at the hash for https://github.com/flutter/flutter/pull/62150, I still see the Dart SDK reporting 2.10.

**Comment by christopherfujino** [comment]
> I haven't been able to reproduce exactly how LUCI is fetching the Dart repository and why the `2.10.0-0.0.dev` tag is showing up in some engine Dart SDK builds but not others.
> 
> I suspect the engine recipe might get more consistent results by doing a `git fetch --tags` instead of just a `git fetch` (see https://flutter.googlesource.com/recipes/+/refs/heads/master/recipes/engine.py#1496)

It's possible the behavior of `git fetch` is not consistent across different versions of git (git api changes have broken tooling before). Explicitly calling `git fetch --tags` would probably be a worthwhile change, though if that explains the inconsistency it wouldn't fix the current version solving situation.

**Comment by zanderso** [comment]
When I comment out the setting of `dart_custom_version_for_pub` here:

https://github.com/flutter/engine/blob/master/tools/gn#L187

Dart reports version:

`2.9.0-edge.<git hash>`

Do we use the custom version string for anything other than looks?

**Comment by jason-simmons** [comment]
The contents of the Dart `version.cc` generated by `make_version.py` will depend on which git tags have been fetched into your local `third_party/dart` repository.

`make_version.py` is selecting the most recenty added `dev` tag (see https://github.com/dart-lang/sdk/blob/master/tools/utils.py#L521)

If you reset back to an engine commit from a long time ago but your local `third_party/dart` has more recent tags, then the `version.cc` will reflect the local tags and not the tag that most closely matches the engine commit's Dart SDK version.

**Comment by dnfield** [comment]
We fetch the tags here: https://flutter.googlesource.com/recipes/+/refs/heads/master/recipes/engine.py#1496

But we should add --tags to it ot make sure it actually gets the tags consistently.

**Comment by a-siva** [comment]
If we change tools/gn in the engine to not setup 
gn_args['dart_custom_version_for_pub'] = 'flutter'
then it would not be doing this git-hash hack and instead report the correct version number.

**Comment by a-siva** [comment]
The tag has been deleted, if a manual roll is attempted again it should get the correct version.

**Comment by christopherfujino** [comment]
> The tag has been deleted, if a manual roll is attempted again it should get the correct version.

We would need to either land a new commit to the engine repo, or re-trigger a build for an existing commit.

**Comment by a-siva** [comment]
> > The tag has been deleted, if a manual roll is attempted again it should get the correct version.
> 
> We would need to either land a new commit to the engine repo, or re-trigger a build for an existing commit.

Is that a problem?

**Comment by christopherfujino** [comment]
> Is that a problem?

Nope, but just to clarify that we will have to roll in a **new** engine build.

**Comment by whesse** [comment]
A safest solution may be to change the Dart SDK so that both 2.9 and 2.10 are accepted as null-safe language versions for auto-enrolled packages (and when the experiment flag is turned on).  That would make all of these failures disappear.  We can release a new dev version of Dart with this behavior, and then all the circular problems disappear.

**Comment by tvolkert** [comment]
I am almost positive that the version hack is no longer used.  @pcsosinski

**Comment by whesse** [comment]
OK, I misunderstood: Nothing is fetching the Dart SDK repo from that tag, Flutter engine is building the pinned SDK, but changing it's version.cc based on tags in the repo, so it is reporting the wrong version to pub resolve, and won't accept the 2.9 packages even though we haven't rolled to 2.10 in the engine?  Can we stop changing the version.cc, and everything will work?

I think there are other issues hit by users on the dev channel, when they do flutter upgrade, but that is a separate issue.

**Comment by a-siva** [comment]
> OK, I misunderstood: Nothing is fetching the Dart SDK repo from that tag, Flutter engine is building the pinned SDK, but changing it's version.cc based on tags in the repo, so it is reporting the wrong version to pub resolve, and won't accept the 2.9 packages even though we haven't rolled to 2.10 in the engine? Can we stop changing the version.cc, and everything will work?
> 
> I think there are other issues hit by users on the dev channel, when they do flutter upgrade, but that is a separate issue.

I have filed https://github.com/dart-lang/sdk/issues/42820 to get rid of that support from the Dart tree.

**Comment by otopba** [comment]
Same issue

**Comment by jakemac53** [comment]
> @tvolkert @jakemac53 Do we use the custom version string for anything other than looks? Does it matter on master?

I think we do check for `flutter` in the version string to detect flutter or not in `build_resolvers` package - if it isn't present `dart:ui` might not be resolvable for build_runner builds.

**Comment by fabiancrx** [comment]
Hitting the same issue while trying to 
`flutter upgrade` on master channel.
```
Building flutter tool...
The current Dart SDK version is 2.10.0-0.0.dev.flutter-9dca49e71e.

Because flutter_tools depends on collection >=1.15.0-nnbd <1.15.0-nullsafety.2 which requires SDK version >=2.9.0-18.0 <=2.9.10, version solving failed.
Error: Unable to 'pub upgrade' flutter tool.
```
Can't output flutter doctor because flutter does not work but I'm from Linux 5.4.51-1-MANJARO x86_64.

**Comment by zanderso** [comment]
@fabiancrx Your existing checkout is in a bad state caused by this issue, so you won't be able to use `flutter upgrade`. However, you should be able to update your checkout directly using `git`. Then running any `flutter` command will first pull a Dart SDK that doesn't have this problem.

**Comment by christopherfujino** [comment]
> Hitting the same issue while trying to
> `flutter upgrade` on master channel.
> 
> ```
> Building flutter tool...
> The current Dart SDK version is 2.10.0-0.0.dev.flutter-9dca49e71e.
> 
> Because flutter_tools depends on collection >=1.15.0-nnbd <1.15.0-nullsafety.2 which requires SDK version >=2.9.0-18.0 <=2.9.10, version solving failed.
> Error: Unable to 'pub upgrade' flutter tool.
> ```
> 
> Can't output flutter doctor because flutter does not work but I'm from Linux 5.4.51-1-MANJARO x86_64.

Assuming you're on `master` branch, `git pull` should get you to the latest (which should have a valid Dart SDK).

**Comment by fabiancrx** [comment]
Thanks @christopherfujino ,@zanderso. A quick `git pull` and then any flutter command such as `flutter -h` does downloads a valid dart SDK (`2.9.0-edge.a3815b659047cfc6b63d41d180aa1ca1b781ee68`).

**Comment by goderbauer** [comment]
It seems this issue is no longer occurring on CI for PRs send to the flutter/flutter repository. However, I just opened one for flutter/packages and I see the issue there again: https://cirrus-ci.com/task/4922239009161216

Is there anything we need to do to make it disappear there as well to unblock PRs?

**Comment by godofredoc** [comment]
Most of the changes were landed on recipes which only affect the luci infrastructure. We may also need to implement the same changes on cirrus.

**Comment by tvolkert** [comment]
FYI, this is now affecting flutter/flutter master (e.g. https://cirrus-ci.com/task/4775934739873792?command=setup#L333), despite us having not rolled a new engine.

**Comment by alsedev** [comment]
Same on Windows 10.
From scratch... install flutter stable from GitHub. 

`git clone https://github.com/flutter/flutter.git -b stable
`
Ok.
`flutter doctor `
Ok.
`flutter channel master (to enable windows desktop test) `
Ok.

Any other flutter command automatically starts downloading Dart SDK 2.10 and...

```
Because flutter_tools depends on collection >=1.15.0-nnbbd <1.15.0-nullsafety.2 which requires SDK version >=2.9.0-18.0 <=2.9.10, version solving failed.
Error (1): Unable to 'pub upgrade' flutter tool.  Retrying in five seconds...

```

Some workaround to this issue?

**Comment by alsedev** [comment]
Trying to downgrade to flutter-1.21-candiate.4 (and 5) but same issue:
```

C:\Dev>git clone https://github.com/flutter/flutter.git -b flutter-1.21-candidate.4
Cloning into 'flutter'...
remote: Enumerating objects: 106, done.
remote: Counting objects: 100% (106/106), done.
remote: Compressing objects: 100% (68/68), done.
remote: Total 250921 (delta 53), reused 46 (delta 35), pack-reused 250815
Receiving objects: 100% (250921/250921), 106.66 MiB | 6.06 MiB/s, done.
Resolving deltas: 100% (192759/192759), done.
Updating files: 100% (4836/4836), done.

C:\Dev>flutter doctor
Checking Dart SDK version...
Downloading Dart SDK from Flutter engine dcc9a4048da890660f0880b4c2afed7c9470d335...
Unzipping Dart SDK...
Building flutter tool...
Running pub upgrade...
The current Dart SDK version is 2.10.0-0.0.dev.flutter-9dca49e71e.

Because flutter_tools depends on collection >=1.15.0-nnbd <1.15.0-nullsafety.2 which requires SDK version >=2.9.0-18.0 <=2.9.10, version solving failed.
Error (1): Unable to 'pub upgrade' flutter tool. Retrying in five seconds... (9 tries left)

Esperando 4294967295, presione CTRL+C para salir ...
Running pub upgrade...
The current Dart SDK version is 2.10.0-0.0.dev.flutter-9dca49e71e.

Because flutter_tools depends on collection >=1.15.0-nnbd <1.15.0-nullsafety.2 which requires SDK version >=2.9.0-18.0 <=2.9.10, version solving failed.
Error (1): Unable to 'pub upgrade' flutter tool. Retrying in five seconds... (8 tries left)

```

**Comment by zanderso** [comment]
@alsedev If you are having the same issue, the best way to let us know that it is important to you is to give a thumbs-up to the issue. As for this one, it looks like the tree is turning green for tip-of-tree of master branch. Thanks!

**Comment by vkpdeveloper** [comment]
I am currently have the same issue after changing flutter channel from stable to master.

**Comment by paps666** [comment]
how can we make this problem the number 1 priority of the flutter team?....this one sucks the most

**Comment by honjian5231** [comment]
This happened to me , checked flutter repo at git , find out there are some pull request ahead at master branch. problem solved.

**Comment by zanderso** [comment]
Tip-of-tree of master should be in a good state, however we are in danger of this happening again until the following two issues are resolved, so I will leave this open.
- [ ] https://github.com/flutter/flutter/issues/55082: "[team] LUCI production builders should never re-upload package that has already been uploaded"
- [ ] https://github.com/dart-lang/sdk/issues/42820 "Delete the '--custom_for_pub' argument from make_version.py" (Or at least remove out-of-band changes to the version string of the Dart SDK that Flutter engine vends.)

Also, please note that this issue is marked P0, and is actively being worked on. If you are still experiencing lingering effects of this issue, your patience is appreciated as we get this cleared up as quickly as we can. Thanks!

**Comment by a-siva** [comment]
with regards to https://github.com/dart-lang/sdk/issues/42820 there is a proposal from @athomas which could solve the problem without the need for out-of-band changes to the Dart SDK version string by Flutter engine.

**Comment by Hixie** [comment]
Is this still P0? And if so, is there anything that needs to happen other than what is covered by https://github.com/flutter/flutter/issues/55082 and https://github.com/dart-lang/sdk/issues/42820?

**Comment by christopherfujino** [comment]
This has been fixed, closing the issue.

**Comment by LisandroDiMeo** [comment]
Maybe someone finds this useful: if you're trying to build for desktop (in linux), don't use **flutter channel dev** use instead **flutter channel master**, this worked for me.

**Comment by jmagman** [comment]
@LisandroDiMeo you are currently seeing this on the dev channel?  What version?

**Comment by LisandroDiMeo** [comment]
> @LisandroDiMeo you are currently seeing this on the dev channel? What version?

![image](https://user-images.githubusercontent.com/22039840/89599416-e8462980-d835-11ea-8fb2-6bca4c08413c.png)

I had to uninstall flutter with snap and reinstall it, after that I switched to master branch.

**Comment by lukepighetti** [comment]
```
cd ~/development/flutter
git checkout master
flutter -v
```

Hope this helps someone

**Comment by christopherfujino** [comment]
Re-opening this as it reared its head again on dev.

**Comment by christopherfujino** [comment]
Apologies about this happening again. What happened was that a bug in our CI code led to a series of engine commits being re-built and re-uploaded, all reporting the latest Dart SDK version. We are working to fix the dev channel. For the time being, you can fix your local checkout by switching to the master channel:

```
git checkout
git pull --ff-only
flutter doctor
```

I will update this issue when the problem has been addressed.

**Comment by christopherfujino** [comment]
This has been resolved by releasing a new dev release, version `1.21.0-7.0.pre`. If you're on the bad dev release and want to upgrade, the following commands should get you back on the happy path:

```
git pull
flutter --version
```

Sorry again for the breakage!

**Comment by lukepighetti** [comment]
This appears to have broken again on `dev` channel again.

**Comment by oGranny** [comment]
I am facing this issue again 
when I run ``flutter create my_app``:

```
flutter create my_app
Error: PowerShell executable not found.
       Either pwsh.exe or PowerShell.exe must be in your PATH.
Creating project my_app...
  my_app\.gitignore (created)
  my_app\.idea\libraries\Dart_SDK.xml (created)
  my_app\.idea\libraries\KotlinJavaRuntime.xml (created)
  my_app\.idea\modules.xml (created)
  my_app\.idea\runConfigurations\main_dart.xml (created)
  my_app\.idea\workspace.xml (created)
  my_app\.metadata (created)
  my_app\android\app\build.gradle (created)
  my_app\android\app\src\main\kotlin\com\example\my_app\MainActivity.kt (created)
  my_app\android\build.gradle (created)
  my_app\android\my_app_android.iml (created)
  my_app\android\.gitignore (created)
  my_app\android\app\src\debug\AndroidManifest.xml (created)
  my_app\android\app\src\main\AndroidManifest.xml (created)
  my_app\android\app\src\main\res\drawable\launch_background.xml (created)
  my_app\android\app\src\main\res\mipmap-hdpi\ic_launcher.png (created)
  my_app\android\app\src\main\res\mipmap-mdpi\ic_launcher.png (created)
  my_app\android\app\src\main\res\mipmap-xhdpi\ic_launcher.png (created)
  my_app\android\app\src\main\res\mipmap-xxhdpi\ic_launcher.png (created)
  my_app\android\app\src\main\res\mipmap-xxxhdpi\ic_launcher.png (created)
  my_app\android\app\src\main\res\values\styles.xml (created)
  my_app\android\app\src\profile\AndroidManifest.xml (created)
  my_app\android\gradle\wrapper\gradle-wrapper.properties (created)
  my_app\android\gradle.properties (created)
  my_app\android\settings.gradle (created)
  my_app\ios\Runner\AppDelegate.swift (created)
  my_app\ios\Runner\Runner-Bridging-Header.h (created)
  my_app\ios\Runner.xcodeproj\project.pbxproj (created)
  my_app\ios\Runner.xcodeproj\xcshareddata\xcschemes\Runner.xcscheme (created)
  my_app\ios\.gitignore (created)
  my_app\ios\Flutter\AppFrameworkInfo.plist (created)
  my_app\ios\Flutter\Debug.xcconfig (created)
  my_app\ios\Flutter\Release.xcconfig (created)
  my_app\ios\Runner\Assets.xcassets\AppIcon.appiconset\Contents.json (created)
  my_app\ios\Runner\Assets.xcassets\AppIcon.appiconset\Icon-App-1024x1024@1x.png (created)
  my_app\ios\Runner\Assets.xcassets\AppIcon.appiconset\Icon-App-20x20@1x.png (created)
  my_app\ios\Runner\Assets.xcassets\AppIcon.appiconset\Icon-App-20x20@2x.png (created)
  my_app\ios\Runner\Assets.xcassets\AppIcon.appiconset\Icon-App-20x20@3x.png (created)
  my_app\ios\Runner\Assets.xcassets\AppIcon.appiconset\Icon-App-29x29@1x.png (created)
  my_app\ios\Runner\Assets.xcassets\AppIcon.appiconset\Icon-App-29x29@2x.png (created)
  my_app\ios\Runner\Assets.xcassets\AppIcon.appiconset\Icon-App-29x29@3x.png (created)
  my_app\ios\Runner\Assets.xcassets\AppIcon.appiconset\Icon-App-40x40@1x.png (created)
  my_app\ios\Runner\Assets.xcassets\AppIcon.appiconset\Icon-App-40x40@2x.png (created)
  my_app\ios\Runner\Assets.xcassets\AppIcon.appiconset\Icon-App-40x40@3x.png (created)
  my_app\ios\Runner\Assets.xcassets\AppIcon.appiconset\Icon-App-60x60@2x.png (created)
  my_app\ios\Runner\Assets.xcassets\AppIcon.appiconset\Icon-App-60x60@3x.png (created)
  my_app\ios\Runner\Assets.xcassets\AppIcon.appiconset\Icon-App-76x76@1x.png (created)
  my_app\ios\Runner\Assets.xcassets\AppIcon.appiconset\Icon-App-76x76@2x.png (created)
  my_app\ios\Runner\Assets.xcassets\AppIcon.appiconset\Icon-App-83.5x83.5@2x.png (created)
  my_app\ios\Runner\Assets.xcassets\LaunchImage.imageset\Contents.json (created)
  my_app\ios\Runner\Assets.xcassets\LaunchImage.imageset\LaunchImage.png (created)
  my_app\ios\Runner\Assets.xcassets\LaunchImage.imageset\LaunchImage@2x.png (created)
  my_app\ios\Runner\Assets.xcassets\LaunchImage.imageset\LaunchImage@3x.png (created)
  my_app\ios\Runner\Assets.xcassets\LaunchImage.imageset\README.md (created)
  my_app\ios\Runner\Base.lproj\LaunchScreen.storyboard (created)
  my_app\ios\Runner\Base.lproj\Main.storyboard (created)
  my_app\ios\Runner\Info.plist (created)
  my_app\ios\Runner.xcodeproj\project.xcworkspace\contents.xcworkspacedata (created)
  my_app\ios\Runner.xcodeproj\project.xcworkspace\xcshareddata\IDEWorkspaceChecks.plist (created)
  my_app\ios\Runner.xcodeproj\project.xcworkspace\xcshareddata\WorkspaceSettings.xcsettings (created)
  my_app\ios\Runner.xcworkspace\contents.xcworkspacedata (created)
  my_app\ios\Runner.xcworkspace\xcshareddata\IDEWorkspaceChecks.plist (created)
  my_app\ios\Runner.xcworkspace\xcshareddata\WorkspaceSettings.xcsettings (created)
  my_app\lib\main.dart (created)
  my_app\my_app.iml (created)
  my_app\pubspec.yaml (created)
  my_app\README.md (created)
  my_app\test\widget_test.dart (created)
  my_app\web\favicon.png (created)
  my_app\web\icons\Icon-192.png (created)
  my_app\web\icons\Icon-512.png (created)
  my_app\web\index.html (created)
  my_app\web\manifest.json (created)
The current Dart SDK version is 2.9.2.

Because my_app depends on flutter any from sdk which requires SDK version >=2.10.0-0.0.dev <3.0.0, version solving failed.
Running "flutter pub get" in my_app...
pub get failed (1; Because my_app depends on flutter any from sdk which requires SDK version >=2.10.0-0.0.dev <3.0.0, version solving
failed.)```
## flutter doctor:

```
[√] Flutter (Channel beta, 1.21.0-9.2.pre, on Microsoft Windows [Version 10.0.18362.1016], locale en-US)
    • Flutter version 1.21.0-9.2.pre at F:\f\mohit\flutter2\flutter
    • Framework revision 81a45ec2e5 (13 days ago), 2020-08-27 14:14:33 -0700
    • Engine revision 20a9531835
    • Dart version 2.9.2


[√] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at f:\f\mohit\Android
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = F:\f\mohit\Android
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Users\bro\AppData\Local\Google\Chrome\Application\chrome.exe

[√] Android Studio (version 4.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 48.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] VS Code (version 1.48.2)
    • VS Code at C:\Users\bro\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.14.1

[√] Connected device (3 available)
    • Web Server (web) • web-server • web-javascript • Flutter Tools
    • Chrome (web)     • chrome     • web-javascript • Google Chrome 85.0.4183.83
    • Edge (web)       • edge       • web-javascript • Microsoft Edge 85.0.564.44

• No issues found!
```

## flutter version:

```
Flutter 1.21.0-9.2.pre • channel beta • https://github.com/flutter/flutter.git
Framework • revision 81a45ec2e5 (13 days ago) • 2020-08-27 14:14:33 -0700
Engine • revision 20a9531835
Tools • Dart 2.9.2
```

please help I wanna use flutter web for my project

**Comment by ElaineSchwner** [comment]
It is happenning on dev on MacOs.

The current Dart SDK version is 2.10.0-4.0.dev.flutter-20bf2fcf56.

Because flutter_tools depends on collection >=1.15.0-nnbd <1.15.0-nullsafety.2 which requires SDK version >=2.9.0-18.0 <=2.9.10, version solving failed.

**Comment by christopherfujino** [comment]
> It is happenning on dev on MacOs.
> 
> The current Dart SDK version is 2.10.0-4.0.dev.flutter-20bf2fcf56.
> 
> Because flutter_tools depends on collection >=1.15.0-nnbd <1.15.0-nullsafety.2 which requires SDK version >=2.9.0-18.0 <=2.9.10, version solving failed.

What Flutter version are you on? This issue was tracking an isolated issue that happened about 2 months ago. If you are on a newer version of flutter, you are experiencing a different issue, and I would appreciate it if you filed a new issue.

**Comment by anandsubbu007** [comment]
Building flutter tool...
Running pub upgrade...
The current Dart SDK version is 2.10.0-4.0.dev.flutter-a3815b6590.

Because flutter_tools depends on collection >=1.15.0-nnbd <1.15.0-nullsafety.2 which requires SDK version >=2.9.0-18.0
  <=2.9.10, version solving failed.
Error (1): Unable to 'pub upgrade' flutter tool. Retrying in five seconds... (9 tries left)

I'm also facing this issue

**Comment by jarlestabell** [comment]
Same happens for me now, after trying to run 'flutter upgrade' on Windows. Now I get this message on all channels.

However, this tip seemed to fix the problem. (Do it in the flutter directory):
`git pull origin --ff-only`

from the first answer in:
https://stackoverflow.com/questions/63204658/version-solving-failed

**Comment by christopherfujino** [comment]
To anyone experiencing this, if you can share the output of the following command, we can diagnose whether this has regressed again or if only individuals on versions that were affected by the original issue are experiencing this:

```
git rev-parse HEAD; git branch --show-current
```

If you ARE experiencing this issue, the fix should be:

```
git fetch; git checkout master; flutter doctor
```

This should get you to a working flutter CLI tool, from which you can then get to your desired channel via the `flutter channel` command.

**Comment by shaan-saz** [comment]
@christopherfujino Facing this issue,
![image](https://user-images.githubusercontent.com/51261384/97963811-ba916200-1ddd-11eb-9946-5a5867caae7c.png)

**Comment by shaan-saz** [comment]
@christopherfujino Facing this issue,

> git rev-parse HEAD -> 
f25bd9c55c48c139524139b477d04b13e9f36b2cdev
git branch --show-current -> dev

**Comment by christopherfujino** [comment]
> @christopherfujino Facing this issue,
> 
> > git rev-parse HEAD ->
> > f25bd9c55c48c139524139b477d04b13e9f36b2cdev
> > git branch --show-current -> dev

Please read the original post, it has instructions for upgrading past.

**Comment by shaan-saz** [comment]
Already solved it, just want to share output of this command as you mentioned in post. Thanks 👍
